client/src/components/MapView.vue

```diff
@@ -179,21 +185,43 @@ export default defineComponent({
 
       // Add geolocate control to the map.
       this.map.addControl(
-        new mapboxgl.GeolocateControl({
-          positionOptions: {
-            // If enabled, gets the best possible results. Can result in slower response times or
-            // increased power consumption so set to false for now.
-            enableHighAccuracy: false,
-          },
-          // Receive updates to the device's location as it changes.
-          trackUserLocation: true,
-        }),
+          new mapboxgl.GeolocateControl({
+            positionOptions: {
+              // If enabled, gets the best possible results. Can result in slower response times or
+              // increased power consumption so set to false for now.
+              enableHighAccuracy: false,
+            },
+            // Receive updates to the device's location as it changes.
+            trackUserLocation: true,
+          }),
       );
 
       // Add zoom in / zoom out buttons to map.
       this.map.addControl(new mapboxgl.NavigationControl());
     },
 
+    /**
+     * Set up listener on the zoom level. This is needed to toggle the data
+     * source for the Lead Connections layer.
+     */
+    setUpZoomListener(): void {
+      if (this.map == null) return;
+
+      this.map.on('zoom', () => {
+        if (this.map == null) return;
```

Comment:
Does 208 not guard against this? Does this give you an error without it?

Reply:
Nope and yes annoyingly. Removed first check and just used null safe ?.